src/components/Campaigns/CampaignDateRange.tsx

```diff
@@ -23,9 +23,9 @@ export const CampaignDateRange = () => {
           value={parseISO(startMeta.value)}
           error={!!startMeta.error}
           helperText={startMeta.error}
-          onChange={(dt) => {
-            startHelper.setValue(formatISO(dt));
-            startHelper.setTouched(true);
+          onChange={async (dt) => {
+            await startHelper.setValue(formatISO(dt));
+            await startHelper.setTouched(true);
```

Comment:
eslint should have detected this dangling promise. in a separate PR could you make sure that eslint is configured to do this like it is on our other projects?